src/cpptools.ts

```diff
@@ -55,19 +55,21 @@ export function parseCompileFlags(args: string[]): CompileFlagInformation {
       extraDefinitions.push(def);
     } else if (value.startsWith('-std=') || lower.startsWith('-std:') || lower.startsWith('/std:')) {
       const std = value.substring(5);
-      if (std.endsWith('++14') || std.endsWith('++1y')) {
-        standard = 'c++14';
-      } else if (std.endsWith('++17') || std.endsWith('++1z') || std.endsWith('++latest')) {
-        standard = 'c++17';
-      } else if (std.endsWith('++11') || std.endsWith('++0x')) {
-        standard = 'c++11';
-      } else if (std.endsWith('++2a')) {
-        // Not yet supported...
-      } else if (std.endsWith('++98')) {
-        standard = 'c++98';
-      } else if (std.endsWith('++03')) {
-        standard = 'c++03';
-      } else {
+      if (lang === 'CXX') {
+        if (std.endsWith('++14') || std.endsWith('++1y')) {
+          standard = 'c++14';
+        } else if (std.endsWith('++17') || std.endsWith('++1z') || std.endsWith('++latest')) {
+          standard = 'c++17';
+        } else if (std.endsWith('++11') || std.endsWith('++0x')) {
+          standard = 'c++11';
+        } else if (std.endsWith('++2a')) {
+          // Not yet supported...
+        } else if (std.endsWith('++98')) {
+          standard = 'c++98';
+        } else if (std.endsWith('++03')) {
+          standard = 'c++03';
+        }
```

Comment:
There is some code a few lines down that deals with "unknown standard control flags".  That code is now incorrect since it will set c++17 for a C file.  It seems this function already sets a default value for `standard` so this is unnecessary, but the warning may still be useful.  Can you replicate the warning for the CXX block as well?